crates/trigger/src/runtime_config/client_tls.rs

```diff
@@ -0,0 +1,41 @@
+use anyhow::Context;
+use rustls_pemfile::private_key;
+use std::io;
+use std::io::Cursor;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "snake_case", tag = "type")]
+pub struct ClientTlsOpts {
+    pub component_ids: Vec<String>,
+    pub hosts: Vec<String>,
+    pub custom_root_ca_file: Option<PathBuf>,
+    pub cert_chain_file: Option<PathBuf>,
+    pub private_key_file: Option<PathBuf>,
+}
+
+// load_certs parse and return the certs from the provided file
+pub fn load_certs(
+    path: impl AsRef<Path>,
+) -> anyhow::Result<Vec<rustls_pki_types::CertificateDer<'static>>> {
+    let contents = fs::read_to_string(path).expect("Should have been able to read the file");
+    let mut custom_root_ca_cursor = Cursor::new(contents);
+
+    Ok(rustls_pemfile::certs(&mut custom_root_ca_cursor)
+        .map(|certs| certs.unwrap())
```

Comment:
As below, this should return an error rather than unwrap.

Reply:
done